frame/dex-router/src/lib.rs

```diff
@@ -0,0 +1,190 @@
+//! # DEX Router Pallet
+//!
+//! Is used to add route to DEX for given asset_id's pair.
+
+#![cfg_attr(not(test), warn(clippy::disallowed_method, clippy::indexing_slicing))] // allow in tests
+#![warn(clippy::unseparated_literal_suffix, clippy::disallowed_type)]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use codec::{Codec, FullCodec};
+	use composable_traits::{
+		defi::LiftedFixedBalance,
+		dex::{CurveAmm, DexRoute, DexRouteNode, DexRouter},
+	};
+	use core::fmt::Debug;
+	use frame_support::pallet_prelude::*;
+	use sp_runtime::{
+		traits::{
+			AtLeast32BitUnsigned, CheckedAdd, CheckedMul, CheckedSub, IntegerSquareRoot, One, Zero,
+		},
+		DispatchResult, FixedPointOperand,
+	};
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		type AssetId: FullCodec
+			+ Eq
+			+ PartialEq
+			+ Copy
+			+ MaybeSerializeDeserialize
+			+ Debug
+			+ Default
+			+ TypeInfo
+			+ Ord;
+		type Balance: Default
```

Comment:
consider using MathBalance as trait bound for part of bounds here

Reply:
Not in this PR. I think that also applied to Dex pallets as well. So I will do it later.

Reply:
thanks